enhancements/scheduling/prevent-user-workloads-from-being-scheduled-on-control-plane-nodes.md

```diff
@@ -229,60 +249,133 @@ spec:
         operator: DoesNotExist
 ```
 
-4. Namespace Labeling for Exceptions:
+4. Service Account to be used by the workload:
 
-- Namespace Administrators are tasked with applying specific labels to their namespaces if exemptions to the default scheduling policies are warranted. This procedure enables selected workloads within these namespaces to bypass the standard restrictions by tolerating the designated NoExecute taints (should be performed on control plane owned namespaces for control plane protection - should be performed on special node group namespaces if that is what needs to be protected).
+- Namespace Administrators or Application Developers need to have a valid Service Account to be used by the workload before the workload is created so the ValidatingAdmisisonPolicy can use the secondary authorization approach above checking if this Service Account can tolerate the relevant taints.
 
 ```
-oc label <NAMESPACE_NAME> openshift.io/control-plane=true
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: special-service-account
+  namespace: default
 ```
 
-5. Pod Scheduling Attempt:
+5. Role and RoleBinding with custom verbs
 
-- Users/Developers proceed to deploy or update pods, embedding scheduling preferences, such as necessary tolerations. Example of adding a relevant toleration:
+- Namespace Administrators or application developers must grant the necessary permissions to this special Service Account to tolerate the relevant taints required. We are using custom verbs and keys, and those are never used by kubernetes itself. We can use the authorizer in the ValidatingAdmissionPolicy to check this custom verbs specific to this workflow.
+
+```
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: tolerate-noexecute-role
+  namespace: default
+# rules: []
+rules:
+- apiGroups: [""]
+  resources: ["node-role.kubernetes.io/control-plane"]
```

Comment:
Are admis/devs expected to use this exact list of NoExecute taints? Or, can they list their subset?

Reply:
Same as above

Reply:
It's on the text (case by case)